pkg/store/postings_codec.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package store
+
+import (
+	"bytes"
+	encoding_binary "encoding/binary"
+	"fmt"
+
+	"github.com/golang/snappy"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/tsdb/index"
+)
+
+const (
+	// these headers should not be prefix of each other
+	codecHeaderRaw    = "diff+varint+raw"
+	codecHeaderSnappy = "diff+varint+snappy"
+)
+
+func isDiffVarintEncodedPostings(input []byte) bool {
+	return bytes.HasPrefix(input, []byte(codecHeaderRaw)) || bytes.HasPrefix(input, []byte(codecHeaderSnappy))
+}
+
+func diffVarintSnappyEncode(p index.Postings) ([]byte, error) {
+	return diffVarintEncode(p, true)
+}
+
+func diffVarintEncode(p index.Postings, useSnappy bool) ([]byte, error) {
```

Comment:
Can we use https://github.com/prometheus/prometheus/blob/cf76daed2f778e97ea70be1bc65ac5f9c5d6d83f/tsdb/encoding/encoding.go#L31 instead of this function?